lib/temporalio/worker/activity_runner.rb

```diff
@@ -12,20 +13,39 @@ class Worker
     #
     # @api private
     class ActivityRunner
-      def initialize(activity_class, start, task_queue, task_token, worker, converter)
+      def initialize(
+        activity_class,
+        start,
+        task_queue,
+        task_token,
+        worker,
+        converter,
+        inbound_interceptors,
```

Comment:
I know we had discussed this before but I forget what your response was. Can you confirm I can write an activity interceptor that _wraps_ all later interceptors? Meaning, can I write an activity interceptor that catches exceptions from activities _including_ other interceptors? If easy enough to writer, can you show me what that looks like? (specifically one interceptor raising out of `execute_activity` and another catching it)

Reply:
Yep, I confirm that you can do that and here's an example:

```ruby
class MyInterceptor
  include Temporalio::Interceptor::ActivityInbound

  def execute_activity(input)
    # here you can modify the input that will be pass down the chain of interceptors
    input.headers['foo'] = 'bar'
    result = yield(input) # you can also just call `super` here, because the default implementation is to yield
    # here you can modify the result, for example assume it's a String and make it upper case
    result.upcase # return value
  rescue => e
    # this will get called if activity or any of the further interceptors raise an error
    logger.error(e)
  end
end
```

Obviously the order of the interceptors matters — in order to inject behaviour around all the interceptors and the implementation, this interceptor will need to be first in the list (it will get called first while traversing into the call and the last when traversing out of it, similar to your typical recursive method).

Reply:
Here's also a spec that demonstrates it (minus the error handling part, which is something I should write a spec for as well) — https://github.com/temporalio/sdk-ruby/blob/main/spec/unit/temporalio/interceptor/chain_spec.rb#L16-L23

Reply:
Perfect, thanks! Definitely makes sense that order matters of course.